### PR TITLE
572 admiral eric

### PR DESCRIPTION
**Pull request by esimms999-gsk**
Minor typos.
As per Stefan: fixed typo in "Parameters" section: should be order = vars(AVISIT, desc(AESEV)) instead of order = desc(AVISIT, desc(AESEV)).
As per Stefan: moved the "Handling of Missing Values" section after the "Parameters" section.
As per Stefan: added example code to Computation section.

### CODE REVIEW

**Review by bundfussr — CHANGES_REQUESTED** [review]
According to the guidance the example codes should be executed and the results be displayed.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/admiral.Rmd [hunk 1]
```diff
@@ -76,6 +76,46 @@ The output dataset is ungrouped. The observations are not ordered in a dedicated
...
+  mutate(TRT01P = ARMCD, TRT01A = ACTARMCD) %>%
+  derive_var_trtsdtm(dataset_ex = ex) %>%
+  mutate(TRTSDT = date(TRTSDTM)) %>%
+  derive_var_trtedtm(dataset_ex = ex) %>%
+  mutate(TRTEDT = date(TRTEDTM)) %>%
+  derive_var_trtdurd() %>%
+  # derive final lab visit date
+  left_join(ds %>% 
+              filter(DSDECOD == "FINAL LAB VISIT") %>%
+              transmute(USUBJID, FINLABDT = convert_dtc_to_dt(DSSTDTC)),
```
[the remark this answers is not in the document]
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd [hunk 2]
```diff
@@ -76,6 +76,46 @@ The output dataset is ungrouped. The observations are not ordered in a dedicated
...
+example below:
+
+```r
+# derive treatment variables
+adsl <- dm %>%
+  mutate(TRT01P = ARMCD, TRT01A = ACTARMCD) %>%
+  derive_var_trtsdtm(dataset_ex = ex) %>%
+  mutate(TRTSDT = date(TRTSDTM)) %>%
+  derive_var_trtedtm(dataset_ex = ex) %>%
+  mutate(TRTEDT = date(TRTEDTM)) %>%
```
[the remark this answers is not in the document]
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd [hunk 3]
```diff
@@ -32,17 +32,23 @@ library(admiral)
...
+```{r, eval=TRUE}
 # derive treatment variables
 adsl <- dm %>%
   mutate(TRT01P = ARMCD, TRT01A = ACTARMCD) %>%
   derive_var_trtsdtm(dataset_ex = ex) %>%
-  mutate(TRTSDT = date(TRTSDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTSDTM)) %>%
   derive_var_trtedtm(dataset_ex = ex) %>%
-  mutate(TRTEDT = date(TRTEDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTEDTM)) %>%
```
Comment (thomas-neitmann) [inline vignettes/admiral.Rmd:44]: You can transform both treatment start and end date in one go: `derive_vars_dtm_to_dt(vars(TRTSDTM, TRTEDTM))`.

File: vignettes/admiral.Rmd [hunk 4]
```diff
@@ -83,22 +89,28 @@ input and return a vector. These function can be used in expressions like
...
+
+# derive treatment variables  
 adsl <- dm %>%
   mutate(TRT01P = ARMCD, TRT01A = ACTARMCD) %>%
   derive_var_trtsdtm(dataset_ex = ex) %>%
-  mutate(TRTSDT = date(TRTSDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTSDTM)) %>%
   derive_var_trtedtm(dataset_ex = ex) %>%
-  mutate(TRTEDT = date(TRTEDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTEDTM)) %>%
```
Comment (thomas-neitmann) [inline vignettes/admiral.Rmd]: Same as above.

File: vignettes/admiral.Rmd [hunk 5]
```diff
@@ -83,22 +89,28 @@ input and return a vector. These function can be used in expressions like
...
   derive_var_trtdurd() %>%
   # derive final lab visit date
-  left_join(ds %>% 
-              filter(DSDECOD == "FINAL LAB VISIT") %>%
+  left_join(ds_final_lab_visit %>% 
               transmute(USUBJID, FINLABDT = convert_dtc_to_dt(DSSTDTC)),
```
Comment (thomas-neitmann) [inline vignettes/admiral.Rmd:107]: Could you move the `transmute()` call up to where you define `ds`?

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/admiral.Rmd (hunk 3 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd (hunk 4 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd (hunk 5 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd [hunk 6]
```diff
@@ -23,7 +23,7 @@ processed dataset. This modular approach makes it easy to adjust code by adding,
 removing, or modifying derivations. Each derivation is a function call. Consider
 for example the following script which creates a (very simple) ADSL dataset.
 
-```r
+```{r, message=FALSE, warning=FALSE}
 library(dplyr)
 library(lubridate)
 library(admiral)
```
Comment (thomas-neitmann) [inline vignettes/admiral.Rmd:29]: Please add `library(cdiscpilot)` here. That's the reason why the checks fail.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/admiral.Rmd [hunk 7]
```diff
@@ -32,17 +32,22 @@ library(admiral)
...
   derive_var_trtsdtm(dataset_ex = ex) %>%
-  mutate(TRTSDT = date(TRTSDTM)) %>%
   derive_var_trtedtm(dataset_ex = ex) %>%
-  mutate(TRTEDT = date(TRTEDTM)) %>%
+  derive_vars_dtm_to_dt(vars(TRTSDTM, TRTEDTM)) %>%
   derive_var_trtdurd()
 ```
 
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(adsl)
```
Comment (bundfussr) [inline vignettes/admiral.Rmd:48]: Please specify the `display_vars` parameter such that `USUBJID` and the new variables created in the example code are displayed.

File: vignettes/admiral.Rmd [hunk 8]
```diff
@@ -76,6 +81,50 @@ The output dataset is ungrouped. The observations are not ordered in a dedicated
...
+  derive_var_trtsdtm(dataset_ex = ex) %>%
+  derive_var_trtedtm(dataset_ex = ex) %>%
+  derive_vars_dtm_to_dt(vars(TRTSDTM, TRTEDTM)) %>%
+  derive_var_trtdurd() %>%
+  # merge on final lab visit date
+  left_join(ds_final_lab_visit, by = "USUBJID")
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(adsl)
```
Comment (bundfussr) [inline vignettes/admiral.Rmd:109]: Please specify the `display_vars` parameter such that `USUBJID` and the new variables created in the example code are displayed.

File: vignettes/admiral.Rmd [hunk 9]
```diff
@@ -76,6 +81,50 @@ The output dataset is ungrouped. The observations are not ordered in a dedicated
...
+```
+# Parameters
```
Comment (bundfussr) [inline vignettes/admiral.Rmd:111]: This is not rendered correctly. I think there needs to be an empty line before a new section.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: vignettes/admiral.Rmd (hunk 6 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd:28]: Added library(admiral.test).

File: vignettes/admiral.Rmd (hunk 7 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd (hunk 8 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

File: vignettes/admiral.Rmd (hunk 9 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd:122]: Fixed.

**Review by thomas-neitmann — APPROVED** [review]

File: vignettes/admiral.Rmd (hunk 9 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd:122]: Done.

Comment (esimms999-gsk) [reply vignettes/admiral.Rmd:122]: Done.

File: vignettes/admiral.Rmd (hunk 8 above)
Comment (esimms999-gsk) [reply vignettes/admiral.Rmd]: Done.

**Comment by thomas-neitmann** [comment]
@bundfussr Could you approve as well? It's currently blocked as you had requested changes in the past.

**Review by bundfussr — APPROVED** [review]

**Pull request merged**